plugins/module_utils/rds.py

```diff
@@ -305,3 +303,69 @@ def ensure_tags(client, module, resource_arn, existing_tags, tags, purge_tags):
             parameters={'ResourceName': resource_arn, 'TagKeys': tags_to_remove}
         )
     return changed
+
+
+def compare_iam_roles(existing_roles, target_roles, purge_roles):
+    '''
+    Returns differences between target and existing IAM roles
+
+        Parameters:
+            existing_roles (list): Existing IAM roles
+            target_roles (list): Target IAM roles
+            purge_roles (bool): Remove roles not in target_roles if True
+
+        Returns:
+            roles_to_add (list): List of IAM roles to add
+            roles_to_delete (list): List of IAM roles to delete
+    '''
+    if target_roles is None:
+        target_roles = []
+    roles_to_add = []
+    roles_to_remove = []
+    for target_role in target_roles:
+        found = False
+        for existing_role in existing_roles:
+            if target_role['role_arn'] == existing_role['role_arn'] and target_role['feature_name'] == existing_role['feature_name']:
+                found = True
+                break
+        if not found:
+            roles_to_add.append(target_role)
+
+    if purge_roles:
+        for existing_role in existing_roles:
+            found = False
+            for target_role in target_roles:
+                if target_role['role_arn'] == existing_role['role_arn'] and target_role['feature_name'] == existing_role['feature_name']:
+                    found = True
+                    break
+            if not found:
+                roles_to_remove.append(existing_role)
+
+    return roles_to_add, roles_to_remove
+
+
+def update_iam_roles(client, module, instance_id, roles_to_add, roles_to_remove):
+    '''
+    Update a DB instance's associated IAM roles
+
+        Parameters:
+            client: RDS client
+            module: AWSModule
+            instance_id: DB's instance ID
+            roles_to_add (list): List of IAM roles to add
+            roles_to_delete (list): List of IAM roles to delete
+
+        Returns:
+            changed (bool): True if changes were successfully made to DB instance's IAM roles; False if not
+    '''
+    for role in roles_to_remove:
```

Comment:
I'm trying to think if there are cases where removing roles form the DB instance before adding roles could result in the credentials being used by the task losing the ability to act on the database.  I believe this only affects service roles but offhand I'm not sure (@gravesm, do you know this?)
It might be worth re-ordering the add_role and remove_role calls.